src/commonMain/kotlin/baaahs/show/live/AutoMode.kt

```diff
@@ -0,0 +1,68 @@
+package baaahs.show.live
+
+import baaahs.control.OpenButtonControl
+import baaahs.doRunBlocking
+import baaahs.util.Logger
+import baaahs.util.globalLaunch
+import kotlin.random.Random
+import kotlinx.coroutines.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.coroutineContext
+
+class AutoMode(
+    private val openShow: OpenShow
+) {
+
+    var indexSeed = 13;
+    var autoMode = false;
+    var running = false;
+    val isRunning: Boolean get() = run {
+        running
+    }
+
+    val autoModeOn: Boolean get() = run {
+        autoMode
+    }
+
+    var gadgetWizard: Job? = null
+
+    fun start() {
+        if (isRunning) return
+        running = true
+        autoMode = true
+        gadgetWizard = CoroutineScope(Dispatchers.Unconfined).launch {
+            playWithButtons()
+        }
+    }
+
+    var frequency = 5000L
+
+    private suspend fun playWithButtons() {
+        val controls: List<OpenButtonControl> = openShow.allFilterButtonControls
+        logger.info { "Controls " + controls.joinToString { it.gadget.title } }
+        val random = Random(indexSeed++)
+        var control: OpenButtonControl
+        while (autoModeOn) {
+            control = controls.random(random)
+            // This doesn't update the UI :(
```

Comment:
Huh, yeah, janky. It definitely seems like `click()` ought to do what you want.

There are a couple ways the button might be enabled/disabled:
* when a user directly clicks on it. In this case, `ButtonControlView` calls `handleToggleClick()`, which calls `button.click()`, but also `onShowStateChange()`, which ultimately calls `ShowManager.Facade.onShowStateChange()`, which re-renders the page.
* when a user on another client clicks it. In this case, I'm not quite sure exactly how the UI gets updated; I _think_ it's something like this: the switch gadget gets an update over pubsub, then notifies its observers, which include `GadgetManager`, which is itself being observed by `StageManager`, which ... somehow magically causes a re-render. I'm missing something there clearly. I'll look more closely in a few.

In short, this is all kinda a spaghetti-tastic mess. Thinking about how to improve matters.